File: public/docs/_examples/dependency-injection/ts/app/heroes/hero.service.ts
// #docregion
import {Injectable} from 'angular2/core';
import {Hero}       from './hero';
import {HEROES}     from './mock-heroes';
import {Logger}     from '../logger.service';

@Injectable()
export class HeroService {
  private _user:string;

  // #docregion internals
  constructor(
    private _logger: Logger,
    private _isAuthorized: boolean) { }

  getHeroes() {
    let auth = this._isAuthorized ? 'authorized ': 'unauthorized';
    this._logger.log(`Getting heroes for ${auth} user.`);
    return HEROES.filter(hero => this._isAuthorized || !hero.isSecret);
  }
  // #enddocregion internals
}

File: public/docs/_examples/dependency-injection/dart/lib/heroes/hero_service.dart
// #docregion
import 'package:angular2/core.dart';

import '../logger_service.dart';
import 'hero.dart';
import 'mock_heroes.dart';

@Injectable()
class HeroService {
  // #docregion internals
  final Logger _logger;
  final bool _isAuthorized;

  HeroService(this._logger, this._isAuthorized);

  List<Hero> getHeroes() {
    var auth = _isAuthorized ? 'authorized' : 'unauthorized';
    _logger.log('Getting heroes for $auth user.');
    return HEROES
        .where((hero) => _isAuthorized || !hero.isSecret)
        .toList();
  }
  // #enddocregion internals
}
